### PR TITLE
Add option to not throw on each failure in summary regression tests

### PR DESCRIPTION
**Pull request by akva2**
Useful for performing further analysis of the failure.

### CODE REVIEW

File: opm/test_util/summaryRegressionTest.cpp [hunk 1]
```diff
@@ -60,7 +62,7 @@ void RegressionTest::getRegressionTest(){
                 if (isRestartFile && keywordString.substr(3,1)=="T"){
                     break;
                 }
-                checkForKeyword(timeVec1, timeVec2, keyword);
+                throwAtEnd = !checkForKeyword(timeVec1, timeVec2, keyword);
```
Comment (joakim-hove) [inline opm/test_util/summaryRegressionTest.cpp]: Shouldn't this be something like `throwAtEnd |= !checkForKeyword( );` to carry a `true` value through the keyword list?

File: opm/test_util/summaryComparator.hpp [hunk 2]
```diff
@@ -26,6 +26,17 @@
 #include <algorithm>
 #include <string>
 
+
+// helper macro to handle error throws or not
+#define THROW_OR_NOT(type, message) \
```
Comment (joakim-hove) [inline opm/test_util/summaryComparator.hpp]: I would suggest `#define ERROR` - the current name seems to indicate that the `THROW_OR_NOT` macro can answer the question: "Should I throw?".

**Comment by joakim-hove** [comment]
This is a welcome addition - personally I would like to see the default behavior to be `throwOnError == false`.

File: opm/test_util/summaryComparator.hpp [hunk 3]
```diff
@@ -26,6 +26,17 @@
...
+  { \
+    if (throwOnError) \
+      OPM_THROW(type, message); \
+    else \
+      std::cout << message << std::endl; \
```
Comment (joakim-hove) [inline opm/test_util/summaryComparator.hpp]: `std::cerr` ?

File: opm/test_util/summaryRegressionTest.cpp (hunk 1 above)
Comment (akva2) [reply opm/test_util/summaryRegressionTest.cpp]: nice catch.

File: opm/test_util/summaryComparator.hpp (hunk 2 above)
Comment (akva2) [reply opm/test_util/summaryComparator.hpp]: which is precisely what this macro does, no ? (through the if()). anyways, i'll rename..

File: opm/test_util/summaryComparator.hpp (hunk 3 above)
Comment (akva2) [reply opm/test_util/summaryComparator.hpp]: indeed. naughty attempt at aiding my scripts, hoped nobody would notice ;)

**Comment by akva2** [comment]
as for throwing by default, i respectfully disagree. it has the potential to spam the terminal a bunch, and i strongly believe that should be opt-in.

**Comment by akva2** [comment]
rebased with fixes.

**Pull request merged**